sklearn/utils/validation.py

```diff
@@ -425,6 +430,14 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         The converted and validated array.
 
     """
+    # warn_on_dtype deprecation check
+    if warn_on_dtype is not None:
+        warnings.warn(
+            "'warn_on_dtype' is deprecated from 'check_array' in "
+            "version 0.20 and will be removed in 0.22. Use "
+            "'warn_on_dtype = None' instead.", DeprecationWarning)
+        warn_on_dtype = None
```

Comment:
Don't set it to `None`: the goal of the deprecation cycle is to preserve the current behaviour for the next 2 versions.